lib/puppet/provider/mysql.rb

```diff
@@ -24,7 +24,7 @@ def self.mysqld_type
     mysqld_version_string.scan(/mariadb/i) { return "mariadb" }
     mysqld_version_string.scan(/\s\(mysql/i) { return "mysql" }
     mysqld_version_string.scan(/\s\(percona/i) { return "percona" }
-    nil
+    return "mysql"
```

Comment:
I believe it's nil specifically for catching unhandled cases instead of propagating incorrectly handled cases. Fail-fast style.

Anyway, I think I agree with @solomonty that if `((Ubuntu))` is the moniker that Ubuntu packagers have chosen to mark their packages with, we can only follow suit.